### PR TITLE
feat: add ListHosts proto

### PR DESCRIPTION
**Pull request by BruceAko**
## Description

This API is added in order to be able to get the number of all hosts stored in the scheduler during E2E testing.

## Related Issue

https://github.com/dragonflyoss/Dragonfly2/issues/3523

### CODE REVIEW

**Review by gaius-qi — CHANGES_REQUESTED** [review]
Added `GetHosts` instead of `GetHostIDs` to return host information.

**Comment by BruceAko** [comment]
> Added `GetHosts` instead of `GetHostIDs` to return host information.

Resolved, please take a look.

**Review by gaius-qi — CHANGES_REQUESTED** [review]

File: pkg/apis/scheduler/v2/scheduler.proto [hunk 1]
```diff
@@ -466,6 +472,9 @@ service Scheduler {
   // AnnounceHost announces host to scheduler.
   rpc AnnounceHost(AnnounceHostRequest)returns(google.protobuf.Empty);
 
+  // GetHosts gets hosts in scheduler.
+  rpc GetHosts(google.protobuf.Empty)returns(GetHostsResponse);
```
[the remark this answers is not in the document]
Comment (BruceAko) [reply pkg/apis/scheduler/v2/scheduler.proto]: Resolved.

**Review by gaius-qi — APPROVED** [review]
LGTM

**Pull request merged**